src/app/store/sagas/watchPort.saga.ts

```diff
@@ -0,0 +1,58 @@
+import * as R from 'ramda';
+import {
+  put,
+  call,
+  take,
+  takeLatest,
+  ActionPattern,
+  cancel
+} from 'redux-saga/effects';
+import createPortChannel from './createPortChannel';
+import { disconnected } from 'app/store/actions/connection';
+import { PortAction } from 'app/store/types';
+
+type MessageSender = browser.runtime.MessageSender;
+type Port = browser.runtime.Port;
+
+const stripReceiverMeta = R.dissocPath<PortAction>(['meta', 'receiver']);
+
+export const createActionForwarder = (port: Port) =>
+  function* forwardAppAction(action: PortAction) {
+    yield call([port, 'postMessage'], stripReceiverMeta(action));
+  };
+
+export const compareMessageSender = (
+  senderA?: MessageSender,
+  senderB?: MessageSender
+): boolean => {
+  return (
+    !!senderA &&
+    !!senderB &&
+    (R.path(['tab', 'id'])(senderA) === R.path(['tab', 'id'])(senderB) ||
+      R.path(['id'])(senderA) === R.path(['id'])(senderB))
+  );
+};
+
+function* watchPortSaga(pattern: ActionPattern, port: Port) {
+  const portChannel = yield call(createPortChannel, port);
+  const forwardActionSaga = yield takeLatest(
+    pattern,
+    createActionForwarder(port)
+  );
+
+  try {
+    while (true) {
+      const channelAction = yield take(portChannel);
+      yield put(channelAction);
+    }
+  } catch (e) {
+    // The error is silenced on purpose. Avoid polluting with an extra disconnected action.
+    // The reason the channel closes is not very useful.
+    // yield put(disconnected(e));
+  } finally {
+    yield cancel(forwardActionSaga);
```

Comment:
Yeah, this is an interesting subject, I agree since I'm silencing the error anyway, but without "cheating" beware that we can't catch the error in the finally block...
I need to verify the following statement, but I think chrome pop the error in the console if there isn't the catch block.

Reply:
Yes, this is why I suggested simply 

```ts
catch () {
  yield cancel(formardActionSaga);
}
```